cli/flox/src/utils/metrics.rs

```diff
@@ -382,6 +382,27 @@ pub trait Connection: Debug + Any + Send + Sync {
     fn into_any(self: Box<Self>) -> Box<dyn Any>;
 }
 
+/// Builds a reqwest client that defaults timeout to TRAILING_NETWORK_CALL_TIMEOUT
+/// and enables in-process hickory_dns on Darwin builds to avoid slow DNS behavior
+/// from Darwin libc's getaddrinfo()
+pub fn build_trailing_reqwest_client() -> Result<reqwest::Client, reqwest::Error> {
+    reqwest::Client::builder()
+        .hickory_dns(cfg!(target_os = "macos") || cfg!(target_os = "ios"))
+        .timeout(TRAILING_NETWORK_CALL_TIMEOUT)
+        .build()
+}
+
+/// Builds a blocking reqwest client that defaults timeout to TRAILING_NETWORK_CALL_TIMEOUT
+/// and enables in-process hickory_dns on Darwin builds to avoid slow DNS behavior
+/// from Darwin libc's getaddrinfo()
+pub fn build_blocking_trailing_reqwest_client() -> Result<reqwest::blocking::Client, reqwest::Error>
+{
+    reqwest::blocking::Client::builder()
+        .hickory_dns(cfg!(target_os = "macos") || cfg!(target_os = "ios"))
+        .timeout(TRAILING_NETWORK_CALL_TIMEOUT)
+        .build()
+}
```

Comment:
This feels like... one way to do this?
I guess I could make a type as well?

I also think it would be better to return a ClientBuilder instead of a Client, but I didn't know what to name the functions. 😅